riscv/src/asm.rs

```diff
@@ -7,58 +7,87 @@ macro_rules! instruction {
         pub unsafe fn $fnname() {
             #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
             core::arch::asm!($asm, $($options)*);
-            #[cfg(not(any(target_arch = "riscv32", target_arch = "riscv64")))]
-            unimplemented!();
```

Comment:
We need `unimplemented!` in case you compile for non-RISC-V targets (e.g., when running tests on x86). The idea is to make sure that, in case you dare to execute `riscv::asm::wfi()` in a test running on x86, the program panics.